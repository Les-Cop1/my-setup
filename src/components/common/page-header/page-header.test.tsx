import React from 'react'

import { PageHeader } from '@components'
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'

describe('Page Header', () => {
  test('PageHeader should have correct background color', () => {
    render(<PageHeader title="default" testid="default" />)
    render(<PageHeader title="white" testid="white" backgroundColor="white" />)
    render(<PageHeader title="light" testid="light" backgroundColor="light" />)
    render(<PageHeader title="dark" testid="dark" backgroundColor="dark" />)
    expect(screen.getByTestId('default')).toHaveClass('bg-white')
    expect(screen.getByText('default')).toHaveClass('text-slate-900')
    expect(screen.getByTestId('white')).toHaveClass('bg-white')
    expect(screen.getByText('white')).toHaveClass('text-slate-900')
    expect(screen.getByTestId('light')).toHaveClass('bg-zinc-50')
    expect(screen.getByText('light')).toHaveClass('text-slate-900')
    expect(screen.getByTestId('dark')).toHaveClass('bg-zinc-800')
    expect(screen.getByText('dark')).toHaveClass('text-white ')
  })

  test('PageHeader should have title', () => {
    render(<PageHeader title="title" testid="default" />)
    expect(screen.getByText('title')).toBeInTheDocument()
  })

  test('Button should have className', () => {
    render(<PageHeader title="title" testid="testId" className="test-class" />)
    expect(screen.getByTestId('testId')).toHaveClass('test-class')
  })

  test('Snapshot', () => {
    const { container } = render(<PageHeader title="Hello" />)
    expect(container).toMatchSnapshot()
  })
})
