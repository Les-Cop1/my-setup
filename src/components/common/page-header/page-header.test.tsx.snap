// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Header Snapshot 1`] = `
<div>
  <div
    class="p-4 sm:p-8 lg:flex lg:items-center lg:justify-between max-w-7xl mx-auto bg-white"
  >
    <div
      class="min-w-0 flex-1"
    >
      <h2
        class="text-2xl font-bold leading-7 sm:truncate sm:text-3xl sm:tracking-tight text-slate-900"
      >
        Hello
      </h2>
    </div>
  </div>
</div>
`;
